src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1356,6 +1390,16 @@ static ucs_status_t uct_ib_mlx5_devx_md_open(struct ibv_device *ibv_device,
                                    UCT_MD_FLAG_INVALIDATE;
             ksm_atomic           = 1;
         }
+
+        if (uct_ib_mlx5dv_check_dm_ksm_reg(md)) {
+            md->flags           |= UCT_IB_MLX5_MD_FLAG_DM_KSM_REG;
+            md->super.cap_flags |= UCT_MD_FLAG_ALLOC;
+        }
+
+        ucs_debug("%s: dm ksm registration %s supported",
+                  uct_ib_device_name(dev),
+                  (md->flags & UCT_IB_MLX5_MD_FLAG_DM_KSM_REG) ? "is" :
```

Comment:
IMO we can make uct_ib_mlx5dv_check_dm_ksm_reg return void, and move these lines (set flag and debug message) inside that function